### PR TITLE
Feature/jsp user profile

### PR DESCRIPTION
**Pull request by adriankulinski**

### CODE REVIEW

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java [hunk 1]
```diff
@@ -57,11 +63,28 @@ public void initRegisterBinder(WebDataBinder binder){
...
+	public Set<String> getHobbiesNames()
+	{
+		Set<String> hobbiesNames = hobbyService.getAllHobbiesNames();
+		return hobbiesNames;
+	}
+	
+	@RequestMapping(value="/profile", method = RequestMethod.GET)
+	public ModelAndView fillProfileView(){
+		ModelAndView model = new ModelAndView("profile");
+		model.addObject(userProfileService.get());
```
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: What does userProfileService.get() return?

Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: UserProfileDTO, method's name is changed

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	public Set<Hobby> getAllHobbies() {
+		Iterator<Hobby> iterator = hobbyRepository.findAll().iterator();
+		Set<Hobby> hobbies = new HashSet<Hobby>();
+	
+		while(iterator.hasNext()){
+			hobbies.add(iterator.next());
+		}
+
```
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java]: Please use streams instead.

Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java]: Sure, done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java [hunk 3]
```diff
@@ -57,11 +63,27 @@ public void initRegisterBinder(WebDataBinder binder){
     public void initChangePasswordBinder(WebDataBinder binder) {
         binder.addValidators(passwordValidator);
     }
+	
+	@ModelAttribute("hobbiesNames")
+	public Set<String> getHobbiesNames()
+	{
+		Set<String> hobbiesNames = hobbyService.getAllHobbiesNames();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: don't assign to variable, just return the set.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/HobbyService.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.pgs.soft.service;
+
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import com.pgs.soft.domain.Hobby;
+
+@Service
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/HobbyService.java]: Please remove that annotation and leave it only on implementation

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+@Service
+public class HobbyServiceImpl implements HobbyService {
+	
+	@Autowired
+	HobbyRepository hobbyRepository;
+
+	@Override
+	public Set<String> getAllHobbiesNames() {
+		Set<String> hobbiesNames = new HashSet<String>();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java]: Please change that method to use stream and map method in it

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java [hunk 6]
```diff
@@ -26,27 +26,58 @@
...
 		userProfile.setStreet(userProfileDTO.getStreet());
 		userProfile.setHomeNumber(userProfileDTO.getHomeNumber());
-		userProfile.setPostCode(userProfileDTO.getPostCode());	
+		userProfile.setPostCode(userProfileDTO.getPostCode());
+		userProfile.getHobbies().clear();
 		
-		for(String hobby : userProfileDTO.getHobbies()){
-			Optional<Hobby> optionalHobby = hobbyRepository.findOneByHobbyName(hobby);
-			optionalHobby.ifPresent(h -> userProfile.getHobbies().add(h));
+		if(userProfileDTO.getHobbies()!=null){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserProfileServiceImpl.java:58]: this check is not needed.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java (hunk 3 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/UserController.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/HobbyService.java (hunk 4 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/HobbyService.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java (hunk 5 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/HobbyServiceImpl.java]: Done.

**Pull request merged**